File: Assets/AltUnityTester/AltUnityDriver/Commands/OldFindObjects/WaitForElementWhereNameContains.cs
public class WaitForElementWhereNameContainsDriver : AltBaseCommand
{
    string name;
    string cameraName;
    bool enabled;
    double timeout;
    double interval;
    public WaitForElementWhereNameContainsDriver(SocketSettings socketSettings, string name, string cameraName, bool enabled, double timeout, double interval) : base(socketSettings)
    {
        this.name = name;
        this.cameraName = cameraName;
        this.enabled = enabled;
        this.timeout = timeout;
        this.interval = interval;
    }
    public AltUnityObject Execute(){
        double time = 0;
        AltUnityObject altElement = null;
        while (time < timeout)
        {
            try
            {
                altElement = new FindElementWhereNameContainsDriver(SocketSettings, name, cameraName, enabled).Execute();
                break;
            }
            catch (System.Exception)
            {
                UnityEngine.Debug.Log("Waiting for element where name contains " + name + "....");
                System.Threading.Thread.Sleep(System.Convert.ToInt32(interval * 1000));
                time += interval;
            }
        }
        if (altElement != null)
            return altElement;
        throw new Assets.AltUnityTester.AltUnityDriver.WaitTimeOutException("Element " + name + " still not found after " + timeout + " seconds");

    }
}

File: Assets/AltUnityTester-BindingsAndExamples(can_be_deleted)/Bindings~/python/altunityrunner/commands/OldFindObjects/wait_for_element_where_name_contains.py
from altunityrunner.commands.command_returning_alt_elements import CommandReturningAltElements
from altunityrunner.commands.OldFindObjects.find_element_where_name_contains import FindElementWhereNameContains
from altunityrunner.altUnityExceptions import WaitTimeOutException
import time
class WaitForElementWhereNameContains(CommandReturningAltElements):
    def __init__(self, socket,request_separator,request_end,appium_driver,value,camera_name='', timeout=20, interval=0.5,enabled=True):
        super().__init__(socket,request_separator,request_end,appium_driver)
        self.value=value
        self.camera_name=camera_name
        self.timeout=timeout
        self.interval=interval
        self.enabled=enabled
    
    def execute(self):
        t = 0
        alt_element = None
        while (t <= self.timeout):
            try:
                alt_element = FindElementWhereNameContains(self.socket,self.request_separator,self.request_end,self.appium_driver,self.value,self.camera_name,self.enabled).execute()
                break
            except Exception:
                print('Waiting for element where name contains ' + self.value + '...')
                time.sleep(self.interval)
                t += self.interval
        if t>=self.timeout:
            raise WaitTimeOutException('Element where name contains ' + self.value + ' not found after ' + str(self.timeout) + ' seconds')
        return alt_element